include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -327,16 +327,15 @@ class kmer_hash_view<urng_t>::basic_iterator
         assert(std::ranges::size(shape_) > 0);
 
         auto urange_size = std::ranges::distance(it_start, it_end);
-        auto step = (shape_.size() + 1 > urange_size) ? 0 : urange_size - shape_.size() + 1;
+        auto step = (shape_.size() > urange_size + 1) ? 0 : urange_size - shape_.size() + 1;
         text_left = std::ranges::next(it_start, step, it_end);
-        text_right = it_end;
```

Comment:
This was an unnecessary assignment

Reply:
Where is text_right defined? Can it be removed?

Reply:
It's a class member, cannot be removed. See `\details` of this constructor as to why we have `it_end`.

But what happened here was:
* `text_right = it_end` at this line
* Maybe calling `hash_full` which immediately does `text_right = text_left`
* `text_right = it_end` at the end